src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -538,7 +538,7 @@ trait Implicits {
     private def checkCompatibility(fast: Boolean, tp0: Type, pt0: Type): Boolean = {
       @tailrec def loop(tp: Type, pt: Type): Boolean = tp match {
         case mt @ MethodType(params, restpe) =>
-          if (mt.isImplicit)
+          if (mt.isImplicit) //
```

Comment:
The crux: this will now unwrap all implicit method types to get to the result type, so the following works:

```
scala> {class A; implicit def bar(implicit b: DummyImplicit)(implicit c: DummyImplicit): A = new A; implicitly[A]} //print

{
  class A;
  implicit def bar(implicit b: scala.Predef.DummyImplicit)(implicit c: scala.Predef.DummyImplicit): A = new A();
  scala.Predef.implicitly[A](bar(Predef.this.DummyImplicit.dummyImplicit)(Predef.this.DummyImplicit.dummyImplicit))
} // : A
```